src/main/java/mat/server/HarpServiceImpl.java

```diff
@@ -114,43 +123,40 @@ public String getHarpClientId() {
         return ServerConstants.getHarpClientId();
     }
 
-//    private TokenIntrospect validate(String token) {
-//        return getClient()
-//                .post()
-//                .uri(uriBuilder -> uriBuilder
-//                        .path("/introspect")
-//                        .queryParam("token", token)
-//                        .queryParam("client_id", ServerConstants.getHarpClientId())
-//                        .queryParam("token_type_hint", "id_token")
-//                        .build())
-//                .accept(MediaType.APPLICATION_JSON)
-//                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
-//                .retrieve()
-//                .bodyToMono(TokenIntrospect.class).block();
-//    }
-//
-//    private ClientResponse revokeToken(String token) {
-//        return getClient()
-//                .post()
-//                .uri(uriBuilder -> uriBuilder
-//                        .path("/revoke")
-//                        .queryParam("token", token)
-//                        .queryParam("client_id", getHarpClientId())
-//                        .queryParam("token_type_hint", "access_token")
-//                        .build())
-//                .accept(MediaType.APPLICATION_JSON)
-//                .contentType(MediaType.APPLICATION_FORM_URLENCODED)
-//                .exchange()
-//                .onErrorResume(e -> Mono.error(new MatException("Unable to revoke token.")))
-//                .block();
-//    }
-//
-//    private WebClient getClient() {
-//        if(isNull(harpOtkaClient)) {
-//            this.harpOtkaClient = WebClient.create(getHarpUrl());
-//        }
-//        return harpOtkaClient;
-//    }
+    @Override
+    public Map<String, String> getUserInfo(String accessToken) {
+        HttpHeaders headers = new HttpHeaders();
+        List<MediaType> acceptList = new ArrayList<>();
+        acceptList.add(MediaType.APPLICATION_JSON);
+        headers.setAccept(acceptList);
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        headers.set("Authorization", "Bearer " + accessToken);
+        HttpEntity<TokenIntrospect> request = new HttpEntity<>(headers);
```

Comment:
Hm... I'm confused. 
HttpEntity of TokenIntrospect type but accepts only headers and there is no body instance of TokenIntrospect?

Reply:
Probably because it's wrong.